File: app/src/main/java/com/kira/mypublishplatform/bean/BaseListBean.java
package com.kira.mypublishplatform.bean;

import java.io.Serializable;
import java.util.List;

public class BaseListBean<T> implements Serializable {

    private int total;
    private List<T> rows;

    public BaseListBean(){}

    public BaseListBean(int count, List<T> data) {
        super();
        this.total = count;
        this.rows = data;
    }

    public int getTotal() {
        return total;
    }

    public void setTotal(int total) {
        this.total = total;
    }

    public List<T> getRows() {
        return rows;
    }

    public void setRows(List<T> rows) {
        this.rows = rows;
    }

}

File: app/src/main/java/com/kira/mypublishplatform/bean/BaseListBean.kt
package com.kira.mypublishplatform.bean

import java.io.Serializable

class BaseListBean<T> : Serializable {
    var total = 0
    var rows: MutableList<T>? = arrayListOf()

    constructor()
    constructor(count: Int, data: MutableList<T>?) : super() {
        total = count
        rows = data
    }

}